File: Mobile/PlatziFlixiOS/PlatziFlixiOS/Data/Entities/RatingDTO.swift
import Foundation

struct RatingRequestDTO: Encodable {
    let userId: Int
    let rating: Int

    enum CodingKeys: String, CodingKey {
        case userId = "user_id"
        case rating
    }
}

struct RatingResponseDTO: Decodable {
    let id: Int
    let courseId: Int
    let userId: Int
    let rating: Int

    enum CodingKeys: String, CodingKey {
        case id
        case courseId = "course_id"
        case userId = "user_id"
        case rating
    }
}

File: Mobile/PlatziFlixAndroid/app/src/main/java/com/espaciotiago/platziflixandroid/data/entities/RatingDTO.kt
package com.espaciotiago.platziflixandroid.data.entities

import com.google.gson.annotations.SerializedName

data class RatingRequestDTO(
    @SerializedName("user_id") val userId: Int,
    @SerializedName("rating") val rating: Int
)

data class RatingResponseDTO(
    @SerializedName("id") val id: Int,
    @SerializedName("course_id") val courseId: Int,
    @SerializedName("user_id") val userId: Int,
    @SerializedName("rating") val rating: Int,
    @SerializedName("created_at") val createdAt: String?,
    @SerializedName("updated_at") val updatedAt: String?
)
